aws-resourceexplorer2-defaultviewassociation/src/main/java/software/amazon/resourceexplorer2/defaultviewassociation/CreateHandler.java

```diff
@@ -51,42 +51,38 @@ public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            AssociateDefaultViewRequest associateDefaultViewRequest = AssociateDefaultViewRequest.builder()
+                    .viewArn(model.getViewArn())
                     .build();
-            } else {
-                logger.log(String.format("[CREATE] A default view is already associated."));
-                return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.AlreadyExists, "A default view is already associated.");
+            AssociateDefaultViewResponse associateDefaultViewResponse;
+            try {
+                associateDefaultViewResponse = proxy.injectCredentialsAndInvokeV2(associateDefaultViewRequest, client::associateDefaultView);
+                // only set the AssociatedAwsPrincipal if the request was successful.
+                model.setAssociatedAwsPrincipal(request.getAwsAccountId());
+                logger.log(String.format("[CREATE] DefaultView created successfully."));
+            } catch (Exception e){
+                HandlerErrorCode thisErrorCode = Convertor.convertExceptionToErrorCode(e, logger);
+                logger.log(String.format("[CREATE] Creating DefaultView failed: %s", thisErrorCode));
+                return ProgressEvent.failed(model, callbackContext, thisErrorCode, "Could not associate a default view: " + e.getMessage());
             }
-        } else {
+            return ProgressEvent.<ResourceModel, CallbackContext>builder()
+                        .resourceModel(model)
+                        .status(OperationStatus.SUCCESS)
+                        .build();
+        } else if (viewArnFromResponse == null) {
             callbackContext.preExistenceCheck = true;
-        }
-
-        AssociateDefaultViewRequest associateDefaultViewRequest = AssociateDefaultViewRequest.builder()
-                .viewArn(model.getViewArn())
-                .build();
-        AssociateDefaultViewResponse associateDefaultViewResponse;
-        try {
-            associateDefaultViewResponse = proxy.injectCredentialsAndInvokeV2( associateDefaultViewRequest, client::associateDefaultView );
-            // only set the AssociatedAwsPrincipal if the request was successful.
-            model.setAssociatedAwsPrincipal(request.getAwsAccountId());
-            logger.log(String.format("[CREATE] DefaultView created successfully."));
-        } catch (Exception e){
-            HandlerErrorCode thisErrorCode = Convertor.convertExceptionToErrorCode(e, logger);
-            logger.log(String.format("[CREATE] Creating DefaultView failed: %s", thisErrorCode));
-            return ProgressEvent.failed(model, callbackContext, thisErrorCode, "Could not associate a default view: " + e.getMessage());
-        }
-
-        return ProgressEvent.<ResourceModel, CallbackContext>builder()
+            return ProgressEvent.<ResourceModel, CallbackContext>builder()
                 .resourceModel(model)
                 .status(OperationStatus.IN_PROGRESS)
```

Comment:
This can happen if for whatever reason callbackContext stays null and if that happens then we will not be able to get retry count also which is part of the context.

Reply:
Ah, you're right, that would require `callbackContext` to be `null` (or missing data), which is its own set of problems. Sorry, carry on!